src/pages/api/index.page.ts

```diff
@@ -61,10 +81,31 @@ export const server = new ApolloServer({
     }
     return {
       user,
+      invalidateCache,
     };
   },
   introspection: true,
-  plugins: [responseCachePlugin({})],
+  plugins: [
+    responseCachePlugin({
+      generateCacheKey: ({ request, queryHash }) => {
+        const geohash = request.variables?.geohash;
+        if (geohash) {
+          return cacheKeyFromGeohash(geohash);
+        }
+        return queryHash;
+      },
+      shouldReadFromCache: async ({ context, cache, request }) => {
+        if (!!context?.user && context?.invalidateCache) {
```

Comment:
then, for most users who are not logged in we always say that they should read from cache.

the exception to this rule are if a user is logged in *and* they have recently set the `LAST_CONTRIBUTION` cookie, meaning that the `invalidateCache` context value is `true`

if this is the case, we remove the implicated tile from the cache and return `false`, meaning that Apollo Server will call the resolver anew to fetch the fresh information